cmd/substreams/run.go

```diff
@@ -17,7 +17,7 @@ import (
 )
 
 func init() {
-	runCmd.Flags().StringP("substreams-endpoint", "e", "mainnet.eth.streamingfast.io:443", "Substreams gRPC endpoint")
+	runCmd.Flags().StringP("substreams-endpoint", "e", "", "Substreams gRPC endpoint")
```

Comment:
Why not make this a breaking change ? 
It will break only on CLI and the fix can be pretty obvious, ex:
`Error: cannot determine network from the manifest and no endpoint specified. Try "-e mainnet.eth.streamingfast.io:443"`

Reply:
I do agree with having our own endpoints as default for specific networks if there is no `-e` override. (ex: `matic -> polygon.streamingfast.io:443`) I don't know who was against that in the start ??

Reply:
isn't the network a required field in the manifest? it should always be defined.  its the endpoint that might not be determined from the network value.  i don't mind doing a print of a deprecation warning (if the network is `mainnet`), but i would prefer a breaking change.

Reply:
Requiring that the network field is populated is harder --> it is a breaking change that prevents people from syncing any spkg created before that change. we must allow an override of the endpoint when the network is not set.

Reply:
`network` is not mandatory. I think we can avoid the breaking change that's why I'm proposing it. Lot's of people are doing today `substreams run <manifest>` with a Substreams manifest that contains no `network`, why breaking them when we can offer a nicer upgrade path.

If everyone thinks breaking is better, I won't resist.